File: libs/m4i-atlas-core/m4i_atlas_core/api/atlas/delete_entity_soft.py
from typing import Optional

from ...entities import EntityMutationResponse
from ..core import atlas_delete

BASE_PATH = "v2/entity/guid"


async def delete_entity_soft(guid: str, access_token: Optional[str] = None):
    """
    Deletes an entity identified by its `guid`. 
    Rather than removing the entity from the database, this API sets the status of the entity to `DELETED`.
    """

    path = f"{BASE_PATH}/{guid}"

    response: str = await atlas_delete(path, access_token=access_token)

    mutations = EntityMutationResponse.from_json(response)

    return mutations
# END delete_entity_soft

File: libs/atlas/api/src/lib/atlas/delete-entity-soft/delete-entity-soft.ts
import { validateRequiredArguments } from '@models4insight/utils';
import { EntityMutationResponse } from '../../types';
import { getHttpClient } from '../atlas-api.module';

const BASE_PATH = 'atlas/v2/entity/guid';

/**
 * Sets the internal status of the entity with the given guid to DELETED. This counts as a soft delete in Atlas.
 * @param guid the guid of the entity
 */
export function deleteEntitySoft(guid: string) {
  const http = getHttpClient();

  validateRequiredArguments(arguments, 'deleteEntitySoft');

  const path = `${BASE_PATH}/${guid}`;

  return http.delete<EntityMutationResponse>(path);
}
